Bump imgui to v1.90.7 and fix pdb installation step

stevewgr opened the pull request:
### Description

Hi @giladreich,

Note that whilst building the project with `/Z7` linker flag, the installation step failed with missing pdb. This is because it assumes the pdb files will always exists on `Debug` or `RelWithDebInfo` builds. Example command:
```sh
cmake .. -G "Visual Studio 17 2022" -A Win32 -DIMGUI_WITH_BACKEND=ON -DIMGUI_BACKEND_PLATFORM=WIN32 -DIMGUI_BACKEND_DX9=ON -DCMAKE_INSTALL_PREFIX=pkg -DCMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$<CONFIG:Debug>:Debug>" -DCMAKE_MSVC_DEBUG_INFORMATION_FORMAT="$<$<CONFIG:Debug,RelWithDebInfo>:Embedded>" -DCMAKE_POLICY_DEFAULT_CMP0141=NEW
```

Setting it to `OPTIONAL` will only install the pdb files if they exists, which should work on all scenarios, even if one configures their Release build to generate symbols, which is not so common. I also improved it by using GNU standard installation dirs using the `GNUInstallDirs` install module.

Additionally I took the courtesy to update the imgui submodule.

stevewgr committed:
Bump imgui submodule from v1.88.0 to v1.90.7.

stevewgr committed:
Fix installation of pdb files to make it optional.

Note that if building the project with /Z7 linker flag, the symbols will
be embedded into the lib and therefore it will fail to install the pdb
files. Setting it to OPTIONAL fixing this problem.

giladreich commented:
Hi @stevewgr, thanks for the update and it also makes sense what you say about the pdb files. Your changes LGTM, but I'll double check once I get home whether any breaking changes or something new added between 1.88 and 1.90.

stevewgr commented:
> Hi @stevewgr, thanks for the update and it also makes sense what you say about the pdb files. Your changes LGTM, but I'll double check once I get home whether any breaking changes or something new added between 1.88 and 1.90.

I didn't notice any breaking changes, but as I'm not so familiar with this library, it will probably be a good idea if you double check as well.

Also a small nit-pick: it will be nice to rename this repo from `cmake-imgui` to `imgui-cmake`. Why? because if I have both imgui and imgui-cmake in the same dir, it sorts them nicely by name, making them easy to find between bunch of other dirs.

stevewgr committed:
Update readme to shallow clone imgui submodule.

Note that imgui repo is quite big. Therefore and to speedup cloning
time, we shallow clone it. Also one can always cd into imgui and git
fetch --unshallow if they need the full git history.

giladreich commented on CMakeLists.txt:
```diff
@@ -3,14 +3,15 @@
 #
 cmake_minimum_required(VERSION 3.15)
 
+include(GNUInstallDirs)
```
This should be under `project` declaration, otherwise cmake will throw a warning:
```
  Unable to determine default CMAKE_INSTALL_LIBDIR directory because no
  target architecture is known.  Please enable at least one language before
  including GNUInstallDirs.
Call Stack (most recent call first):
  CMakeLists.txt:6 (include)
```

stevewgr replied:
Done: https://github.com/giladreich/cmake-imgui/pull/7/commits/40a0c05ccbe572e5527756887388acf3b3895904

giladreich reviewed (approved):
LGTM.

> Also a small nit-pick: it will be nice to rename this repo from cmake-imgui to imgui-cmake. Why? because if I have both imgui and imgui-cmake in the same dir, it sorts them nicely by name, making them easy to find between bunch of other dirs.

I'll keep that in mind 👍 I have some local branch (not finished yet) that allows building all imgui backends on all supported platforms. That will refactor the entire way of using this module, so perhaps I'll also use it as an opportunity to rename the repo.

Thanks again.
